index.js

```diff
@@ -1,264 +0,0 @@
-const startButton = document.getElementById('start-button');
-const restartButton = document.getElementById('restart-button');
-const startScreen = document.getElementById('game-intro');
-const gameOverScreen = document.getElementById('game-over');
-let scoreGameOver = document.getElementById('score');
-
-let canvas;
-let imgForest;
-let imgDragon;
-let imgEnemy;
-let imgSheep;
-let imgMessage;
-let dragon = {
-    x: 300,
-    y: 480,
-    w: 83,
-    h: 110
-}
-
-let enemiesArray = [];
-let fireballsArray = [];
-let sheepArray = [];
-let scoreCounter = 0;
-let sheepSpawnBuffer;
-let sheepSpawnDifficulty;
-let enemySpawnBuffer;
-let enemySpawnDifficulty;
-
-let fireballSound;
-let gameOverSound;
-let battleMusic;
-
-
-function preload() {
-    imgForest = loadImage('img/bg-blurred-with-frame.png');
-  
-    imgDragon = loadImage('img/dragons-head.png');
-
-    imgEnemy = loadImage('img/knight.png');
-
-    imgSheep = loadImage('img/sheep.png');
-
-    imgMessage = loadImage('img/message-frame.png');
-
-    fireballSound = loadSound('audio/fireball-sound.wav');
-
-    gameOverSound = loadSound('audio/game-over.wav');
-
-    battleMusic = loadSound('audio/battle-music.wav');
-
-  } // end of preload function
-
-  function setup() {
-    canvas = createCanvas(1200, 600);
-    canvas.hide();
-    // gameOverScreen.style.display = "none";
-    noLoop();
-
-  } // end of setup function
-
-  function draw() {
-    background(220);
-    image(imgForest, 0, 0, 1200, 600);
-    image(imgDragon, mouseX - 50, dragon.y, dragon.w, dragon.h);
-    
-    // score counter while game on
-  textFont('Optima');
-  fill(241, 226, 200);
-  textSize(32);
-  text('Score: ', 240, 40, 20, 40);
-  text(scoreCounter, 340, 40, 20, 40);
-
-  fireballShot();
-  enemyBreach();
-  stopFireballs();
-  sheepPointsUp();  
-  levelUp();
-
-  sheepSpawnBuffer++;
-  if (sheepSpawnBuffer % sheepSpawnDifficulty === 1) {
-    createSheep();
-  }
-
-  enemySpawnBuffer++;
-  if (enemySpawnBuffer % enemySpawnDifficulty === 1) {
-    createEnemy();
-  }
-
-  for (enemy of enemiesArray) {
-    enemy.y += 2;
-
-    image(imgEnemy, enemy.x, enemy.y, enemy.w, enemy.h);
-    noFill();
-    noStroke();
-    rect(enemy.x, enemy.y, enemy.w, enemy.h);
-  }
-
-  for (sheep of sheepArray) {
-    sheep.y += 2;
-
-    image(imgSheep, sheep.x, sheep.y, sheep.w, sheep.h);
-    noFill();
-    noStroke();
-    rect(sheep.x, sheep.y, sheep.w, sheep.h);
-  }
-    
-  for (fireball of fireballsArray) {
-        fireball.y -= 15;   // speed of fireballs
-        fill(255,0,0);
-        stroke(255, 215, 0);
-        circle(fireball.x, fireball.y, 20);
-        
-  }
-
-}   // end of draw function
-
-
-
-
-// fireball-knight collision: when a fireball touches the enemy it'll remove both, the fireball and the enemy and add a point to score
-function fireballShot() {
```

Comment:
I think these function names could be better, the purpose of this function could be made more obvious by calling it something like checkForFireballCollision()